src/mlte/validation/condition.py

```diff
@@ -0,0 +1,119 @@
+"""
+The interface for measurement validation.
+"""
+from __future__ import annotations
+
+import typing
+from typing import Callable, Any
+import base64
+
+import dill
+
+from mlte.value import Value
+from . import Result
+
+
+class Condition:
+    """
+    The Condition class defines the interface for measurement validators.
+    """
+
+    @typing.no_type_check
+    def __init__(
+        self,
+        name: str,
+        arguments: list[Any],
+        callback: Callable[[Value], Result],
+    ):
+        """
+        Initialize a Condition instance.
+
+        :param name: The name of the name method, for documenting purposes.
+        :type name: str
+        :param callback: The callable that implements validation
+        :type callback: Callable[[Value], Result]
+        """
+
+        self.name: str = name
+        """The human-readable identifier for the name method."""
+
+        self.arguments: list[Any] = arguments
+        """The arguments used when validating the condition."""
+
+        self.callback: Callable[[Value], Result] = callback
+        """The callback that implements validation."""
+
+    def __call__(self, value: Value) -> Result:
+        """
+        Invoke the validation callback
+
+        :param value: The value of measurement evaluation
+        :type value: Value
+
+        :return: The result of measurement validation
+        :rtype: Result
+        """
+        return self.callback(value)._with_evidence_metadata(value.metadata)
+
+    def to_json(self) -> dict[str, Any]:
+        """Returns this requirement as a dictionary."""
+        """        try:
+            validator = getattr(value, self.condition)
+        except AttributeError:
+            raise RuntimeError(
+                f"Invalid validation method provided: '{self.condition}()' method not found for value of type {value.typename}"
+            )
+        condition: Condition = validator(self.threshold)"""
+        return {
+            "name": self.name,
+            "arguments": self.arguments,
```

Comment:
Perhaps I missed this elsewhere in the review, but what are these `arguments`? Are we sure that they can be serialized?

Reply:
That is a very good question. I mentioned it briefly above, but these are the one or many arguments passed over to the condition thunk when validating. Typically it is the threshold. These arguments are stored in the callback, and as thus are implicitly stored in the thunk that is dilled into the JSON file. So this field is not really required. My main reasoning to add it was to leave documentation of the arguments passed into the condition, since they end up being hidden in the callback, and there is no easy way to recover them and see what was passed to it after loading a Spec. So if a condition is .less_than(3), the name of the condition, "less_than", is stored in the spec, but the "3" is only stored dilled in the callback, so there is no way to see it. For a user to be able to see "less_than" what, we need to store it in a separate field. This field is actually not used in any way other than maybe to show this potentially in a UI, or report, or when examining the Spec. So I believe it is important to have it.

Reply:
That being said, I am not sure they can be serialized. In practice, the arguments we have for the current validators are always primitives to they are serializable, but in theory new conditions could use non-serializable arguments. I opened ticket #150 for this.